src/main/java/fr/uge/dummychargepoint/input/Type.java

```diff
@@ -1,8 +1,17 @@
 package fr.uge.dummychargepoint.input;
 
+import fr.uge.dummychargepoint.exception.IllegalTypeException;
+import java.util.Locale;
+
 public enum Type {
 
-  CLIENT,
-  SERVER
+  CLIENT, SERVER;
 
+  public static Type fromString(String string) {
+    try {
+      return Type.valueOf(string.toUpperCase(Locale.ROOT));
+    } catch (IllegalArgumentException e) {
+      throw new IllegalTypeException("Wrong type given, must be CLIENT or SERVER.", e);
+    }
+  }
```

Comment:
The message should not be here, we should throw the exception, catch in the main/service, then tell why is it happening.

Here on a business logic, the enum should just say that he doesn't know what type is it.

it could be:

```suggestion
  public static Type fromString(String string) {
    try {
      return Type.valueOf(string.toUpperCase(Locale.ROOT));
    } catch (IllegalArgumentException e) {
      throw new IllegalTypeException("Unknown type", e);
    }
  }
```

Reply:
Why not? We are in the enum, and it knows what values are acceptable am I wrong?

Reply:
I don't really know to be honest, the two ways are good for me...

Let's go with your way since we are not creating a service (spring boot) but an CLI app